### DISCUSSION TITLE
Master builds unable to access secrets

### DISCUSSION THREAD

**Discussion by JoshKeegan** [community/community#47581] [Actions]
### Select Topic Area

Bug

### Body

I have a Github Actions pipeline that uses a secret set on the repository to log in to docker hub. Here's the [yaml](https://github.com/JoshKeegan/docker-zip/blob/ac4a5239f38d472c70342bce6ba3302b207c7b27/.github/workflows/cicd.yaml). 
This docker login step only runs on master, so it should always have access to the secret, and this used to work.

I have not changed the definition, but builds have started failing when the commit is merging a PR, e.g. [failed run](https://github.com/JoshKeegan/docker-zip/actions/runs/4166049314/jobs/7255490400).
If I commit to master, trigerring the CI it works, e.g. [successful run](https://github.com/JoshKeegan/docker-zip/actions/runs/4186653862/jobs/7255439254).

This is frustrating because it means for all PRs dependabot raises, the build after it is merged to master now fails.

**Comment by unknowVariable** [comment]
It sounds like the issue is related to the environment variables that are being used to authenticate with Docker Hub. It's possible that when a pull request is merged, the environment variables are not being set correctly.

To troubleshoot the issue, you can try the following:

1. Check that the DOCKERHUB_TOKEN secret is still set in the repository secrets. You can do this by going to the repository's settings page and clicking on "Secrets".

2. Double-check that the DOCKERHUB_TOKEN secret is correctly defined in the workflow file. It should be defined in the with section of the docker/login-action step.

3. Make sure that the DOCKERHUB_TOKEN secret is set as a secret environment variable in the repository's settings. To do this, go to the repository's settings page, click on "Environments", and then click on the "Secrets" tab.

4. Add some debug output to your workflow file to print out the value of the DOCKERHUB_TOKEN secret when the workflow is run. This can help you to verify that the secret is being correctly set. For example, you could add the following step to your workflow file:
`- name: Print DOCKERHUB_TOKEN
  run: echo ${{ secrets.DOCKERHUB_TOKEN }}`

5. If the above steps do not resolve the issue, you could try using a personal access token (PAT) instead of the DOCKERHUB_TOKEN secret to authenticate with Docker Hub. This would involve creating a new PAT with the necessary permissions, adding it as a secret to the repository, and then modifying the docker/login-action step to use the new secret. This could help to rule out any issues with the DOCKERHUB_TOKEN secret.

I hope this helps you to resolve the issue with your GitHub Actions pipeline!

**Reply by JoshKeegan** [reply]
Thanks for the quick response!

> It sounds like the issue is related to the environment variables that are being used to authenticate with Docker Hub. It's possible that when a pull request is merged, the environment variables are not being set correctly.
> 
> To troubleshoot the issue, you can try the following:
> 
>     1. Check that the DOCKERHUB_TOKEN secret is still set in the repository secrets. You can do this by going to the repository's settings page and clicking on "Secrets".
Yes, It's still there:
![image](https://user-images.githubusercontent.com/1682204/219118767-c2a49b38-fec3-4d06-935d-9561d63bee5b.png)

> 
>     2. Double-check that the DOCKERHUB_TOKEN secret is correctly defined in the workflow file. It should be defined in the with section of the docker/login-action step.
Yes, [it's specified in the with section](https://github.com/JoshKeegan/docker-zip/blob/ac4a5239f38d472c70342bce6ba3302b207c7b27/.github/workflows/cicd.yaml#L23). This yaml file has also not been changed since September, and builds have been working up until now.
> 
>     3. Make sure that the DOCKERHUB_TOKEN secret is set as a secret environment variable in the repository's settings. To do this, go to the repository's settings page, click on "Environments", and then click on the "Secrets" tab.
The repository has no environments:
![image](https://user-images.githubusercontent.com/1682204/219119258-20edc98f-0841-4034-bfbd-f51008a7f40a.png)

>     4. Add some debug output to your workflow file to print out the value of the DOCKERHUB_TOKEN secret when the workflow is run. This can help you to verify that the secret is being correctly set. For example, you could add the following step to your workflow file:
>        `- name: Print DOCKERHUB_TOKEN   run: echo ${{ secrets.DOCKERHUB_TOKEN }}`
In the login of the failed run you can see it is not present without any debug print:
![image](https://user-images.githubusercontent.com/1682204/219119927-b1986c52-1bc7-4263-b40e-73e9ff7f9534.png)

> 
>     5. If the above steps do not resolve the issue, you could try using a personal access token (PAT) instead of the DOCKERHUB_TOKEN secret to authenticate with Docker Hub. This would involve creating a new PAT with the necessary permissions, adding it as a secret to the repository, and then modifying the docker/login-action step to use the new secret. This could help to rule out any issues with the DOCKERHUB_TOKEN secret.
I have the same problem in another repositories too, [e.g. dotnet-build](https://github.com/JoshKeegan/dotnet-build/actions/runs/4178249208). It also works if I commit, so the token is working.
> 
> 
> I hope this helps you to resolve the issue with your GitHub Actions pipeline!

**Reply by stevenvegt** [reply]
We have the same issue. Yesterday the docker build action stopped working. Nothing changed in the secrets or action config file. In the action log, we used to see the `username` and `password` fields under the `with` object, now they don't:

<img width="547" alt="Screenshot 2023-02-15 at 17 42 58" src="https://user-images.githubusercontent.com/3896736/219310606-e78650fe-92ea-4773-bb31-82b831cda74e.png">
<img width="383" alt="Screenshot 2023-02-15 at 17 42 46" src="https://user-images.githubusercontent.com/3896736/219310617-282b6a10-13e4-405e-acde-ac09c2eb5bbf.png">

Action config file is here: https://github.com/nuts-foundation/nuts-node/blob/master/.github/workflows/build-images.yaml
Secrets are set in the repository under `Actions secrets and variables` -> `Repository secrets`.

**Reply by stevenvegt** [reply]
Ah, found it :) In our case these were dependabot PRs merged by a comment containing `@dependabot merge` instead of merging it by hand (e.g. clicking on the merge button). Probably something to do with dependabot being an owner of the commit and therefor not having access to the secrets?

**Reply by JoshKeegan** [reply]
> In our case these were dependabot PRs merged by a comment containing `@dependabot merge` instead of merging it by hand 

:eyes: That's probably the same as I'm getting. Next time I get a dependabot PR on this repo I'll merge it by hand and see if that works.

If that is the case though, it seems like a bug. If a change is merged to master (either manually or via `@dependabot merge`) I'd expect that master build to have access to the secrets, and up until recently it did.

**Comment by jkowalski** [comment]
Seeing the same thing in github.com/kopia/kopia. Basically when action runs after push to the master branch it does not get access to secrets if the commit was made by dependabot, but works fine if the commit was made by a human. 

This looks like a regression and basically breaks our test builds. For me the last time commit triggered by dependabot had access to regular secrets was on 4/13 and on 4/14 was the first occurrence when a build in master did not get secrets.
